File: apps/go/application/fizz_buzz_value_command.go
package application

import (
	type_ "github.com/k2works/getting-started-tdd/apps/go/domain/type_"
)

// FizzBuzzValueCommand は単一の FizzBuzzValue を生成するコマンドです。
type FizzBuzzValueCommand struct {
	number       int
	fizzBuzzType type_.FizzBuzzType
}

// NewFizzBuzzValueCommand は FizzBuzzValueCommand を生成します。
func NewFizzBuzzValueCommand(number int, fizzBuzzType type_.FizzBuzzType) *FizzBuzzValueCommand {
	return &FizzBuzzValueCommand{number: number, fizzBuzzType: fizzBuzzType}
}

// Execute は値生成コマンドを実行し FizzBuzzValue を返します。
func (c *FizzBuzzValueCommand) Execute() interface{} {
	return c.fizzBuzzType.Generate(c.number)
}

File: apps/rust/src/application/fizz_buzz_value_command.rs
use crate::domain::types::FizzBuzzType;
use crate::fizz_buzz::create;

use super::FizzBuzzCommand;

pub struct FizzBuzzValueCommand {
    fizz_buzz_type: Box<dyn FizzBuzzType>,
    number: i32,
}

impl FizzBuzzValueCommand {
    pub fn new(type_number: i32, number: i32) -> Result<Self, String> {
        let fizz_buzz_type = create(type_number)?;
        Ok(Self {
            fizz_buzz_type,
            number,
        })
    }
}

impl FizzBuzzCommand for FizzBuzzValueCommand {
    fn execute(&self) -> String {
        self.fizz_buzz_type.generate(self.number).to_string()
    }
}
